hypershift-operator/controllers/nodepool/apiserver-haproxy/haproxy.cfg

```diff
@@ -26,4 +26,4 @@ backend remote_apiserver
   option httpchk GET /version
   option log-health-checks
   default-server inter 10s fall 3 rise 3
-  server controlplane {{ .ExternalAPIAddress }}:{{ .ExternalAPIPort }}
+  server controlplane {{ .RemoteIP }}:{{ .RemotePort }}
```

Comment:
This looks like it's just pointing at the Service ClusterIP - if you were to learn the actual api server pod IPs via DNS service discovery and a headless service ([this](https://www.securityandit.com/network/haproxy-for-service-discovery-in-kubernetes/) blog post covers it pretty well) you could then have haproxy actually balance traffic across the API server pods more intelligently.

Reply:
For future changes :)